src/testdir/test_functions.vim

```diff
@@ -951,6 +951,11 @@ func Test_mode()
     call assert_equal('nt', mode(1))
     call feedkeys(":echo \<C-R>=Save_mode()\<C-U>\<CR>", 'xt')
     call assert_equal("c-c", g:current_modes)
+    call feedkeys(":\<insert>\<C-r>=Save_mode()\<CR>",'xt')
```

Comment:
These shouldn't be inside this `if has('terminal')` block. I think they should be moved just below line 920.

Reply:
i am sorry for making so many mistakes.